src/main/java/org/jabref/gui/fieldeditors/AbstractEditorViewModel.java

```diff
@@ -65,8 +65,10 @@ public void bindToEntry(BibEntry entry) {
                 fieldBinding,
                 newValue -> {
                     if (newValue != null) {
-                        // Controlsfx uses hardcoded \n for multiline fields, but JabRef stores them in OS Newlines format
-                        String oldValue = entry.getField(field).map(value -> value.replace(OS.NEWLINE, "\n").trim()).orElse(null);
+                        // A file may be loaded using CRLF. ControlsFX uses hardcoded \n for multiline fields.
+                        // Normalizing is done during writing of the .bib file (see org.jabref.logic.exporter.BibWriter.BibWriter).
+                        // Thus, we need to normalize the line endings.
+                        String oldValue = entry.getField(field).map(value -> value.replace("\r\n", "\n").trim()).orElse(null);
```

Comment:
OS.Newline please as mac only uses \r

Reply:
I thought, that was Mac OS 9 and earlier: Source https://stackoverflow.com/a/13295575/873282. Mac OS 9 was [EOL in 2001](https://de.wikipedia.org/wiki/Mac_OS_9).

More sources: https://de.wikipedia.org/wiki/Zeilenvorschub#Computer links to https://de.wikipedia.org/wiki/Mac_OS_(Classic) for `\r` as line separator.

Please double check if we really need to support `\r`. I would really consider it being outdated!

---

My case illustrated by `crlf.bib` is as follows:

- A Windows user edits a file using their favourit editor. 
- The editor stores CRLF
- The file is sent to a Linux user
- The Linux user opens the file and positions the cursor in the comment field
- Autosave triggers
- Cursor moves (this is described in the comments why we need these checks)

If I do not overlook something:  `OS.Newline` does not help: OS.Newline is `\n` on Linux. The file uses `\r\n` (because it was created on Windows). - In other words: OS.Newline is dependent on the current operating system and not on the operating system the .bib file was created.

I could route through `org.jabref.model.database.BibDatabase#getNewLineSeparator` to `org.jabref.gui.fieldeditors.AbstractEditorViewModel#bindToEntry`. This seems to be too much effort for the gain that `\r\n` is supported.

Reply:
Yes, but Excel export to CSV still uses CR. :-P 
Then go ahead with crlf

Reply:
For reference: https://answers.microsoft.com/en-us/msoffice/forum/all/csv-line-endings/2b4eedaf-22ef-4091-b7dc-3317303d2f71

In case we ever encounter this, we can use git magic to auto-fix: https://nicercode.github.io/blog/2013-04-30-excel-and-line-endings/